include/sound/sof/topology.h

```diff
@@ -57,8 +57,8 @@ struct sof_ipc_comp {
 	uint32_t pipeline_id;
 	uint32_t core;
 
-	/* reserved for future use */
-	uint32_t reserved[1];
+	/* extended data length, 0 if no extended data */
+	uint32_t ext_data_length;
```

Comment:
I am not a big fan of running out of options if we need a new flag...

Reply:
The fact is that it is used up, if we need a new flag, we should bump the ABI major per my understanding. @kv2019i This is an common scenario of ABI management you proposed?

Reply:
> 64KB will require 32bits, but yes, I think using only 16bits is sufficient for us.

Reply:
> @keyonjie @plbossart @marc-hb Given we are adding a new extension mechanism that can be used to add more fields and the more minor consideration that 32bit values are easier for the DSP to handle, this seems ok to me.
> Plus firmware is already merged, so this conversation is a bit late at this point.

Yes, I think we finally aligned on using all the 32 bits.